ui_advanced_config.py

```diff
@@ -284,325 +430,233 @@ def BuildUI(self):
+            "<<ComboboxSelected>>", self.change_objects_entry_widgets
         )
+        self.show_object_entry(self.currentObjectData)
 
+        # MAP
         self.mapData = self.ldr.map_templates
         self.mapIDs = self.ldr.getMapIDs()
+        print(self.mapIDs)
+        self.currentMapData = self.mapData[self.mapIDs[0]]
+
         self.selectMapsCombo.configure(values=self.mapIDs)
         self.selectMapsCombo.current(0)
-        self.selectMapsCombo.bind("<<ComboboxSelected>>", self.updateMapsEntryWidgets)
-        self.mapsIDEntry.insert(0, self.mapIDs[0])
-        self.mapsNameEntry.insert(0, self.mapData[self.mapIDs[0]].getData("name"))
-        self.mapsEdgeObjIDEntry.insert(
-            0, self.mapData[self.mapIDs[0]].getData("edge_obj_id")
-        )
-        self.mapsDescEntry.insert(0, self.mapData[self.mapIDs[0]].getData("desc"))
-        self.mapsWidthEntry.insert(0, self.mapData[self.mapIDs[0]].getData("width"))
-        self.mapsHeightEntry.insert(0, self.mapData[self.mapIDs[0]].getData("height"))
-
-    def updateTeamEntryWidgets(self, v):
-        currentTeamIdx = self.selectTeamCombo.current()
-        self.teamNameEntry.delete(0, tk.END)
-        self.teamNameEntry.insert(0, self.teamNames[currentTeamIdx])
-        self.teamSizeEntry.delete(0, tk.END)
-        self.teamSizeEntry.insert(0, self.teamData[self.teamNames[0]]["size"])
-        self.callsignEntry.delete(0, tk.END)
-        self.callsignEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["callsign"]
-        )
-        self.squadEntry.delete(0, tk.END)
-        self.squadEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["squad"]
-        )
-        self.agentObjectEntry.delete(0, tk.END)
-        self.agentObjectEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["object"]
-        )
-        self.aiFileEntry.delete(0, tk.END)
-        self.aiFileEntry.insert(
-            0, self.teamData[self.teamNames[0]]["agent_defs"][0]["AI_file"]
+        self.selectMapsCombo.bind(
+            "<<ComboboxSelected>>", self.change_maps_entry_widgets
         )
+        self.show_map_entry(self.currentMapData)
+
+    def change_team_entry_widgets(self, event=None):
+        """
+        Gets the correct team data for the currently selected team.
+        """
+        # the answer variable defaults to true
+        self.answer = True
+
+        # if any of the team entry values differ from their starting values,
+        # the user is warned that they could be overwritten
+        if not (
+            (
+                (self.teamNameEntry.get() == self.currentTeamData["name"])
+                and (int(self.teamSizeEntry.get()) == self.currentTeamData["size"])
+                and (
+                    self.callsignEntry.get()
+                    == self.currentTeamData["agent_defs"][0]["callsign"]
+                )
+            )
+        ):
+            self.answer = askyesno(
```

Comment:
I don't get this error if I make a change, don't update, and switch teams.

Reply:
Should be resolved in my next commit.

Reply:
LGTM

Reply:
Resolved